synapse/http/servlet.py

```diff
@@ -287,6 +288,8 @@ class attribute containing a pre-compiled regular expression. The automatic
     into the appropriate HTTP response.
     """
 
+    PERMISSION_CODE = None
```

Comment:
I feel like the whole `PERMISSION_CODE` thing is a bit opaque, compared with just passing the desired permission code into `validate_admin_token`.